code/crates/config/src/lib.rs

```diff
@@ -339,6 +342,208 @@ mod gossipsub {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(tag = "load_type", rename_all = "snake_case")]
+pub enum MempoolLoadType {
+    UniformLoad(UniformLoadConfig),
+    NoLoad,
+    NonUniformLoad(NonUniformLoadConfig),
+}
+
+impl Default for MempoolLoadType {
+    fn default() -> Self {
+        Self::NonUniformLoad(NonUniformLoadConfig::default())
+    }
+}
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(from = "nonuniformload::RawConfig", default)]
```

Comment:
Why not derive Deserialize directly on this structure instead of passing by a `RawConfig`?

Reply:
I saw that it was done like that for `GossipSubConfig` so I thought that was a standard.

Reply:
Ah I see! This was only done for GossipSubConfig to make sure we would go through it's constructor in order to fine-tune the parameters